### DISCUSSION TITLE
Mediatr vs Event Bus

### DISCUSSION THREAD

**Discussion by oluatte** [kgrzybek/modular-monolith-with-ddd#152] [Q&A] [answered]
I am trying to understand the usage of Mediatr vs the custom event bus implementation in the code. 

It seems like mediatr could handle both cases in which they are used that is, Mediatr for domain events and the Event Bus for integration events. 

Any thoughts on why this done?

**Comment by itinicore** [comment]
That's only true because it's in the  same project. Once you want to separate one module from the project, for example for scalability reasons MediatR is no longer valid option.

**Reply by oluatte** [reply]
Hey @itinicore, I totally get that. 

What I'm hoping to do here is move gradually towards isolated modules. At the moment I have zero users (still building) and it might be quite a while (or never) until I have enough users to require isolated modules. 

Until then, I would like to be have a simpler implementation while the modules are together and in memory. This is what spurred my question. How do i do the simple thing first and the more robust/complex thing later?

**Comment by csalat** [comment, marked as answer]
mass transit mediator

**Reply by oluatte** [reply]
Thanks @csalat. That's a fantastic suggestion. I am checking it now and hoping that the Mediator can easily be upgraded/swapped to use durable messages.
